PG12 build error - varlena

PG12 disallows varlen arrays as it pretends everyone lives in C90.

```
gis@geocint:~$ sudo pgxn install https://github.com/bytesandbrains/h3-pg/archive/master.zip
INFO: saving /tmp/tmpaj2yzmzu/master
INFO: unpacking: /tmp/tmpaj2yzmzu/master
INFO: building extension
git clone --branch v3.6.0 --depth 1 \
        https://github.com/uber/h3 libh3-v3.6.0
Cloning into 'libh3-v3.6.0'...
remote: Enumerating objects: 292, done.
remote: Counting objects: 100% (292/292), done.
remote: Compressing objects: 100% (279/279), done.
remote: Total 292 (delta 43), reused 119 (delta 7), pack-reused 0
Receiving objects: 100% (292/292), 15.59 MiB | 1.36 MiB/s, done.
Resolving deltas: 100% (43/43), done.
Note: checking out '7728307a4b43dc0179c336ee35527c1d043723a1'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

mkdir -p libh3-v3.6.0/build
cd libh3-v3.6.0/build && cmake \
        -DCMAKE_C_FLAGS=-fPIC \
        -DBUILD_TESTING=OFF \
        -DENABLE_COVERAGE=OFF \
        -DENABLE_DOCS=OFF \
        -DENABLE_FORMAT=OFF \
        -DENABLE_LINTING=OFF \
        ..
-- The C compiler identification is GNU 9.2.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
CMake Warning at CMakeLists.txt:248 (message):
  clang-format was not detected, so automatic source code reformatting is
  disabled


CMake Warning at CMakeLists.txt:260 (message):
  clang-tidy was not detected, so source code linting is disabled


-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build
cmake --build libh3-v3.6.0/build --target h3
make[1]: Entering directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
make[2]: Entering directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
make[3]: Entering directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
make[4]: Entering directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
Scanning dependencies of target h3
make[4]: Leaving directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
make[4]: Entering directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
[  9%] Building C object CMakeFiles/h3.dir/src/h3lib/lib/algos.c.o
[  9%] Building C object CMakeFiles/h3.dir/src/h3lib/lib/coordijk.c.o
[ 18%] Building C object CMakeFiles/h3.dir/src/h3lib/lib/bbox.c.o
[ 27%] Building C object CMakeFiles/h3.dir/src/h3lib/lib/polygon.c.o
[ 27%] Building C object CMakeFiles/h3.dir/src/h3lib/lib/h3Index.c.o
[ 36%] Building C object CMakeFiles/h3.dir/src/h3lib/lib/vec2d.c.o
[ 45%] Building C object CMakeFiles/h3.dir/src/h3lib/lib/vec3d.c.o
[ 45%] Building C object CMakeFiles/h3.dir/src/h3lib/lib/linkedGeo.c.o
[ 54%] Building C object CMakeFiles/h3.dir/src/h3lib/lib/localij.c.o
[ 63%] Building C object CMakeFiles/h3.dir/src/h3lib/lib/geoCoord.c.o
[ 63%] Building C object CMakeFiles/h3.dir/src/h3lib/lib/h3UniEdge.c.o
[ 72%] Building C object CMakeFiles/h3.dir/src/h3lib/lib/mathExtensions.c.o
[ 81%] Building C object CMakeFiles/h3.dir/src/h3lib/lib/vertexGraph.c.o
[ 90%] Building C object CMakeFiles/h3.dir/src/h3lib/lib/faceijk.c.o
[ 90%] Building C object CMakeFiles/h3.dir/src/h3lib/lib/baseCells.c.o
[100%] Linking C static library lib/libh3.a
make[4]: Leaving directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
[100%] Built target h3
make[3]: Leaving directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
make[2]: Leaving directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
make[1]: Leaving directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
cmake --build libh3-v3.6.0/build --target binding-functions
make[1]: Entering directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
make[2]: Entering directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
make[3]: Entering directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
make[4]: Entering directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
Scanning dependencies of target binding-functions
make[4]: Leaving directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
make[4]: Entering directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
make[4]: Leaving directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
Built target binding-functions
make[3]: Leaving directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
make[2]: Leaving directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
make[1]: Leaving directory '/tmp/tmpaj2yzmzu/h3-pg-master/libh3-v3.6.0/build'
sed -e 's/@EXTVERSION@/unreleased/g' \
        src/extension.in.h > src/extension.h
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -Ilibh3-v3.6.0/build/src/h3lib/include -I. -I./ -I/usr/include/postgresql/12/server -I/usr/include/postgresql/internal  -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/hierarchy.o src/hierarchy.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -Ilibh3-v3.6.0/build/src/h3lib/include -I. -I./ -I/usr/include/postgresql/12/server -I/usr/include/postgresql/internal  -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/miscellaneous.o src/miscellaneous.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -Ilibh3-v3.6.0/build/src/h3lib/include -I. -I./ -I/usr/include/postgresql/12/server -I/usr/include/postgresql/internal  -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/type.o src/type.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -Ilibh3-v3.6.0/build/src/h3lib/include -I. -I./ -I/usr/include/postgresql/12/server -I/usr/include/postgresql/internal  -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/uniedges.o src/uniedges.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -Ilibh3-v3.6.0/build/src/h3lib/include -I. -I./ -I/usr/include/postgresql/12/server -I/usr/include/postgresql/internal  -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/extension.o src/extension.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -Ilibh3-v3.6.0/build/src/h3lib/include -I. -I./ -I/usr/include/postgresql/12/server -I/usr/include/postgresql/internal  -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/indexing.o src/indexing.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -Ilibh3-v3.6.0/build/src/h3lib/include -I. -I./ -I/usr/include/postgresql/12/server -I/usr/include/postgresql/internal  -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/opclass_hash.o src/opclass_hash.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -Ilibh3-v3.6.0/build/src/h3lib/include -I. -I./ -I/usr/include/postgresql/12/server -I/usr/include/postgresql/internal  -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/traversal.o src/traversal.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -Ilibh3-v3.6.0/build/src/h3lib/include -I. -I./ -I/usr/include/postgresql/12/server -I/usr/include/postgresql/internal  -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/inspection.o src/inspection.c
src/inspection.c: In function ‘h3_get_faces’:
src/inspection.c:96:2: error: ISO C90 forbids variable length array ‘elements’ [-Werror=vla]
   96 |  Datum  elements[maxFaces];
      |  ^~~~~
cc1: some warnings being treated as errors
make: *** [<builtin>: src/inspection.o] Error 1
ERROR: command returned 2: make PG_CONFIG=/usr/bin/pg_config all

```

--- a/src/inspection.c
+++ b/src/inspection.c
@@ -93,11 +93,12 @@ h3_get_faces(PG_FUNCTION_ARGS)
 	int			maxFaces = maxFaceCount(*hex);
 
 	ArrayType  *result;
-	Datum		elements[maxFaces];
+	
 	int			nelems = 0;
 
 	/* get the faces */
 	int		   *faces = palloc(maxFaces * sizeof(int));
+	Datum		*elements = palloc(maxFaces * sizeof(Datum));
 
 	h3GetFaces(*hex, faces);
 

